qiime2/metadata.py

```diff
@@ -14,6 +14,38 @@
 
 class Metadata:
     def __init__(self, dataframe):
+        # `/` and `\0` aren't permitted because they are invalid filename
+        # characters on *nix filesystems. The remaining values aren't permitted
+        # because they *could* be misinterpreted by a shell (e.g. `*`, `|`).
+        illegal_chars = ['/', '\0', '\\', '*', '<', '>', '?', '|', '$']
+        illegal_chars = set(illegal_chars)
+        chars_for_msg = ", ".join("%r" % i for i in illegal_chars)
+
+        categories = dataframe.columns.values.tolist()
+        indices = dataframe.index.tolist()
+
+        if dataframe.columns.dtype_str != 'object':
```

Comment:
oh I see! What you have is fine then, but I think the dtype could also be `str` and that would be valid as well. This probably would only happen when someone initializes the object from the constructor (instead of `.load` or `.from_artifact`).